### PR TITLE
feat: add all app settings

### PR DESCRIPTION
**Pull request by peterdeme**

### CODE REVIEW

**Comment by peterdeme** [comment]
We should refactor this with some inheritance logic with the next major...

File: app.go [hunk 1]
```diff
@@ -7,19 +7,30 @@ import (
...
-	AutoTranslationEnabled *bool               `json:"auto_translation_enabled,omitempty"`
-	Grants                 map[string][]string `json:"grants,omitempty"`
-	MigratePermissionsToV2 *bool               `json:"migrate_permissions_to_v2,omitempty"`
-	PermissionVersion      string              `json:"permission_version,omitempty"`
-	FileUploadConfig       *FileUploadConfig   `json:"file_upload_config,omitempty"`
-	ImageUploadConfig      *FileUploadConfig   `json:"image_upload_config,omitempty"`
+	DisableAuth               *bool               `json:"disable_auth_checks,omitempty"`
+	DisablePermissions        *bool               `json:"disable_permissions_checks,omitempty"`
+	APNConfig                 *APNConfig          `json:"apn_config,omitempty"`
+	FirebaseConfig            *FirebaseConfig     `json:"firebase_config,omitempty"`
```
Comment (ferhatelmas) [inline app.go:13]: while we're here, we could add huawei/xiaomi configs too

File: app.go [hunk 2]
```diff
@@ -37,13 +51,13 @@ func (a *AppSettings) SetAPNConfig(c APNConfig) *AppSettings {
 	return a
 }
 
-func (a *AppSettings) SetFirebaseConfig(c FirebaseConfig) *AppSettings {
+func (a *AppSettings) SetFirebaseConfig(c FirebaseConfigRequest) *AppSettings {
```
Comment (ferhatelmas) [inline app.go:54]: this separation is a good direction but beware that this is a breaking change

Comment (peterdeme) [reply app.go:54]: Should we raise major? I don’t know how many people uses this

**Review by ferhatelmas — APPROVED** [review]

File: app.go (hunk 2 above)
Comment (ferhatelmas) [reply app.go:54]: I am fine with big major number as much as we're semantic

**Pull request merged**